src/realm/object-store/object_store.cpp

```diff
@@ -633,7 +634,16 @@ static void create_initial_tables(Group& group, std::vector<SchemaChange> const&
         }
         void operator()(AddIndex op)
         {
-            add_search_index(table(op.object), *op.property, op.type);
+            switch (op.type) {
```

Comment:
This should presumably be in `add_search_index()` rather than duplicating it in each of the cases?

Reply:
add_search_index actually already did the right thing and we just weren't calling it right in the additive schema change case.